src/libAtomVM/nifs.c

```diff
@@ -2466,3 +2491,281 @@ static term nif_console_print(Context *ctx, int argc, term argv[])
     }
     return OK_ATOM;
 }
+
+static char b64_table[64] = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
+
+// per https://tools.ietf.org/rfc/rfc4648.txt
+
+static term base64_encode(Context *ctx, int argc, term argv[], bool return_binary)
+{
+    UNUSED(argc);
+    term src = argv[0];
```

Comment:
this will reference invalid memory due to `memory_ensure_free` which is called later

Reply:
Got it -- argv[0] used in changes.